bpfman/src/multiprog/xdp.rs

```diff
@@ -329,7 +329,12 @@ impl XdpDispatcher {
         });
         for (i, v) in extensions.iter_mut().enumerate() {
             let id = v.0.get_program_id()?;
-            let mut ext = Extension::from_pin(format!("{RTDIR_FS}/prog_{id}"))?;
+            let mut ext = Extension::from_pin(format!("{RTDIR_FS}/prog_{id}")).map_err(|e| {
+                BpfmanError::LoadPinError {
+                    path: format!("{RTDIR_FS}/prog_{id}"),
+                    reason: e,
+                }
+            })?;
```

Comment:
I wonder what value we'd really get from an additional error variant here. We don't have anything that really matches/switches on these variants; we propagate errors back through main and print whatever `Error` happens to be. The `LoadPinError` variant would be justified if the code base had error recovery logic that matched on specific variants. But, even then, the variant name should reflect that it's a program that failed to load (e.g., `PinnedProgramLoadError`).

```suggestion
            let path = format!("{RTDIR_FS}/prog_{id}");
            let mut ext = Extension::from_pin(&path).map_err(|e| {
                BpfmanError::Error(format!("Unable to load pinned BPF program from {}: {}", path, e))
            })?;
```

Reply:
If you're happy with the suggestion then I think we should report path in all the other cases too.

Reply:
AFAICT the code with `extension::from_pin` is only in `tc.rs` and `xdp.rs`
I added a few other minor changes (added helper func `prog_path` instead of the redundant format! statement and I saw that ` initialize_bpfman() -> anyhow::Result<()> {` could benefit from some path logging as well. However, I may be introducing too many dumb changes (don't fix it if it ain't broken ?) so lmk